criu/mem.c

```diff
@@ -132,12 +139,27 @@ u64 should_dump_page(pmc_t *pmc, VmaEntry *vmae, u64 vaddr, bool *softdirty)
 		}
 		if (vaddr < pmc->regs[pmc->regs_idx].start)
 			return pmc->regs[pmc->regs_idx].start;
+
+		if (pmc->regs[pmc->regs_idx].categories & PAGE_IS_GUARD) {
+			if (guard)
+				*guard = true;
+
+			return vaddr + PAGE_SIZE;
+		}
+
 		if (softdirty)
 			*softdirty = pmc->regs[pmc->regs_idx].categories & PAGE_IS_SOFT_DIRTY;
 		return vaddr;
 	} else {
 		u64 pme = pmc->map[PAGE_PFN(vaddr - pmc->start)];
 
+		if (vma_entry_is(vmae, VMA_ANON_PRIVATE) && (pme & PME_GUARD_REGION)) {
```

Comment:
are you sure that it will be always VMA_ANON_PRIVATE?

Reply:
sigh. While I was working on this had a local kernel copy from February, but now I see that https://github.com/torvalds/linux/commit/f807123d578df4218e2580a1e1bb3436f4567c4a was added. So it was not possible to have anything except anonymous private mappings, but now it is possible...

Thanks for pointing this out!